spec/models/piece_spec.rb

```diff
@@ -50,12 +50,64 @@
       expect(move.game_id).to eq(@game.id)
     end
 
-    it "should raise an error if the arguments are out of bounds" do
-      piece = @game.pieces.create(
-        row: 0, col: 0, is_captured: false, user: @user
-      )
+    # MeO tests for move_to! - test valid_move? and its called methods
+    from_row = 3
+    from_col = 2
+
+    it "should not move the piece: nil move" do
```

Comment:
I anticipate that the else branch will go away for production. It just contains my logger message for now.

I see your point - so I added explicit tests for the private methods move_nil?, move_out_of_bounds? and valid_move? for a nil move and an out-of-bounds move. I removed the move_to test for a nil move since it doesn't prove anything.